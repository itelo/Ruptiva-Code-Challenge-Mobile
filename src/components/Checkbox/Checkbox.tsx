import React from "react";
import { TouchableOpacity, Text, GestureResponderEvent } from "react-native";
import { ThemeType } from "@utils/styles/ThemeContext";
import useStyles from "@utils/styles/useStyles";
import Check from "../Check";

type CheckBoxProps = {
  touchableTestID?: string
  labelTestID?: string;
  checked: boolean;
  label: string;
  onPress(event: GestureResponderEvent): void;
  disabled?: boolean
}

const CheckBox = (props: CheckBoxProps) => {
  const styles = useStyles(classes, props);

  return (
    <TouchableOpacity
      testID={props.touchableTestID}
      style={styles.buttonContainer}
      onPress={props.onPress}
      disabled={props.disabled}
    >
      <Check checked={props.checked} />
      <Text style={styles.label} testID={props.labelTestID}>
        {props.label}
      </Text>
    </TouchableOpacity>
  )
};

CheckBox.defaultProps = {
  disabled: false
};

const classes = (theme: ThemeType, props: CheckBoxProps) => ({
  buttonContainer: {
    flexDirection: "row" as "row",
    alignItems: "center" as "center"
  },
  label: {
    marginLeft: 8,
    ...theme.typography.body2,
  }
})

export default CheckBox;
