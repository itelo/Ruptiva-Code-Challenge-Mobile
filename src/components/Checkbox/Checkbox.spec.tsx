import React from "react";
import Checkbox from "./Checkbox";
import { render, fireEvent } from "react-native-testing-library";
import renderer from "react-test-renderer";

describe("<Checkbox />", () => {
  it("fire onPress event", () => {
    const onPressMock = jest.fn();
    const { getByTestId } = render(
      <Checkbox
        checked
        touchableTestID="button"
        onPress={onPressMock}
        label="button"
      />
    );

    const button = getByTestId("button");

    fireEvent.press(button);
    expect(onPressMock).toHaveBeenCalledTimes(1);
  });

  it("fire onPress event and change the checked value", () => {
    let checked = false;
    const onPressMock = () => {
      checked = !checked;
    };
    const { getByTestId, update, toJSON } = render(
      <Checkbox
        checked={checked}
        touchableTestID="button"
        onPress={onPressMock}
        label="button"
      />
    );

    const button = getByTestId("button");
    expect(toJSON()).toMatchSnapshot();

    fireEvent.press(button);

    update(<Checkbox
      checked={checked}
      touchableTestID="button"
      onPress={onPressMock}
      label="button"
    />)

    expect(toJSON()).toMatchSnapshot();
  });

  it("test props checked={false}", () => {
    const onPressMock = jest.fn();
    const { toJSON } = render(
      <Checkbox
        checked={false}
        onPress={onPressMock}
        label="button"
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it("test props checked={true}", () => {
    const onPressMock = jest.fn();
    const { toJSON } = render(
      <Checkbox
        checked={false}
        onPress={onPressMock}
        label="button"
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});