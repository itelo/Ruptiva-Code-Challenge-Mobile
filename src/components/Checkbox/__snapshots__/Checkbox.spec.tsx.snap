// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> fire onPress event and change the checked value 1`] = `
<View
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "opacity": 1,
    }
  }
  testID="button"
>
  <View
    style={
      Object {
        "borderColor": "#00b2ff",
        "borderRadius": 10,
        "borderWidth": 1,
        "height": 20,
        "width": 20,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 10,
            "height": 14,
            "margin": 2,
            "width": 14,
          },
          Object {
            "backgroundColor": "#FFF",
          },
        ]
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "SF Pro Text",
        "fontSize": 14,
        "fontWeight": "400",
        "letterSpacing": 0.17136,
        "lineHeight": 21,
        "marginLeft": 8,
      }
    }
  >
    button
  </Text>
</View>
`;

exports[`<Checkbox /> fire onPress event and change the checked value 2`] = `
<View
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "opacity": 1,
    }
  }
  testID="button"
>
  <View
    style={
      Object {
        "borderColor": "#00b2ff",
        "borderRadius": 10,
        "borderWidth": 1,
        "height": 20,
        "width": 20,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 10,
            "height": 14,
            "margin": 2,
            "width": 14,
          },
          Object {
            "backgroundColor": "#00b2ff",
          },
        ]
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "SF Pro Text",
        "fontSize": 14,
        "fontWeight": "400",
        "letterSpacing": 0.17136,
        "lineHeight": 21,
        "marginLeft": 8,
      }
    }
  >
    button
  </Text>
</View>
`;

exports[`<Checkbox /> test props checked={false} 1`] = `
<View
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "borderColor": "#00b2ff",
        "borderRadius": 10,
        "borderWidth": 1,
        "height": 20,
        "width": 20,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 10,
            "height": 14,
            "margin": 2,
            "width": 14,
          },
          Object {
            "backgroundColor": "#FFF",
          },
        ]
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "SF Pro Text",
        "fontSize": 14,
        "fontWeight": "400",
        "letterSpacing": 0.17136,
        "lineHeight": 21,
        "marginLeft": 8,
      }
    }
  >
    button
  </Text>
</View>
`;

exports[`<Checkbox /> test props checked={true} 1`] = `
<View
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "borderColor": "#00b2ff",
        "borderRadius": 10,
        "borderWidth": 1,
        "height": 20,
        "width": 20,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 10,
            "height": 14,
            "margin": 2,
            "width": 14,
          },
          Object {
            "backgroundColor": "#FFF",
          },
        ]
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "SF Pro Text",
        "fontSize": 14,
        "fontWeight": "400",
        "letterSpacing": 0.17136,
        "lineHeight": 21,
        "marginLeft": 8,
      }
    }
  >
    button
  </Text>
</View>
`;
